src/udev/net/link-config.c

```diff
@@ -380,15 +381,15 @@ int link_config_apply(link_config_ctx *ctx, link_config *config,
 
         speed = DIV_ROUND_UP(config->speed, 1000000);
 
-        r = ethtool_set_glinksettings(&ctx->ethtool_fd, old_name, speed, config->duplex, config->autonegotiation);
+        r = ethtool_set_glinksettings(&ctx->ethtool_fd, old_name, config);
```

Comment:
You no longer pass speed to ethtool_set_glinksettings(), and you used config->speed inside ethtool_set_glinksettings() instead of DIV_ROUND_UP(config->speed, 1000000)